site/docs/types/compound_logical_types.md

```diff
@@ -1,15 +1,15 @@
-# Compound Types
+# Compound Logical Types
```

Comment:
Okay, that explains a lot (like the filename).

I do think there should be a word to distinguish "type" from the "~~logical~~ type", though. One of the things that went very wrong in the Arrow Substrait consumer is the lack of an understanding that a Substrait type is actually a four-tuple of the ~~logical~~ type, nullability, an optional variation, and a list of parameters (I know the latter doesn't actually exist for simple types, but in my head and in the validator it's just an empty set of parameters). I used "base type" in the validator sources, would naming them that way make more sense to you?

Reply:
I think we should talk about things like: 
* All types have certain properties
    * Can have multiple physical variations
    * Can be nullable or non-nullable.
* Some types:
    * have integer parameters
    * have parameters composed of other types.

I understand the concern and confusion. I'm not sure it warrants two terms because there aren't really two separate type concepts. 

If we are looking for a comparison, a more apt comparison to code may be that we have type classes and type instances. For example, I can have I8?[0] and I8[1]. They are both instances of the I8 type class. However, that feels like it is talking about implementation of the spec, not the spec itself. I8 does not exist independent of its definition that includes nullability and variation within the spec.

Reply:
Just to be clear, I was just looking for a term to use for the first part of a type, as in:

```
        data type
   _________|________
   STRUCT ? [0] <...>
   ‾‾|‾‾‾ | ‾|‾ ‾‾|‾‾
    ???   |  |  parameter pack
          |  |
nullability  variation
```

(using your syntax; I'm still confused about the [0] part, but that's more suited for another thread in this list)

Type class is fine; incidentally that's also what I used internally in the validator. I'm fine with whatever. I simply used "logical type" for it in the proposal without any further intended implications, so s/logical type/type class/g (I'll get around to rewriting it based on the feedback here when I feel like I'm clear on everything again).

Reply:
I'd use "name" for the first part I guess.(?)

Reply:
Meh... The `STRUCT` part is more than just a name; it also encapsulates the semantics of the type and what the parameter pack (if any) means. "User-defined type name" also feels overly broad to me.

Do you have anything against type class? I suppose it's a bit of an overloaded term from object-oriented type systems, but at least Substrait's type system isn't using the word "class" for anything else that I'm aware of.

Reply:
I'm ok with class.

Reply:
By the way, nice little ascii graphic above :D